src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1933,6 +1946,50 @@ BlockIO InterpreterCreateQuery::fillTableIfNeeded(const ASTCreateQuery & create)
             /* async_isnert */ false).execute();
     }
 
+    /// If the query is a CREATE TABLE .. CLONE AS ..., attach all partitions of the source table to the newly created table.
+    if (create.is_clone_as && !as_table_saved.empty() && !create.is_create_empty && !create.is_ordinary_view && !create.is_live_view
+        && (!(create.is_materialized_view || create.is_window_view) || create.is_populate))
+    {
+        String as_database_name = getContext()->resolveDatabase(create.as_database);
+        StorageID as_table_id = {as_database_name, as_table_saved};
+        StoragePtr as_table = DatabaseCatalog::instance().tryGetTable(as_table_id, getContext());
+        if (!as_table)
+        {
+            return {};
```

Comment:
Should we throw an exception in this case?

Reply:
I already checked if the source table is a MergeTree family table at another place. At this point, the new table is created. So I think we consider the source table deleted or removed when cloning the data.

Reply:
I think it's still better to throw an exception in this case. Or we can save the StoragePtr that we have checked before, reuse it here, and avoid this check at all

Reply:
I just realized that `ATTACH PARTITION ALL` is already supported, so I don't need to get the partition_ids from the source table, and this code segment can be removed.